core/core-impl/src/main/java/ai/timefold/solver/core/api/solver/ProblemStatistics.java

```diff
@@ -0,0 +1,12 @@
+package ai.timefold.solver.core.api.solver;
+
+/**
+ * The statistics of a given problem submitted to a {@link Solver}.
+ *
+ * @param entityCount The number of genuine entities defined by the problem.
+ * @param variableCount The number of genuine variables defined by the problem.
+ * @param maximumValueRangeSize The number of possible assignments for the genuine variable with the largest range.
```

Comment:
We already have to iterate over every entity to get this number. So I'm wondering... why not replace this number with a total number of distinct values?

Could be a problem for int value ranges, which are uncountable. How do you handle that case right now?

Reply:
Currently, `maximumValueRangeSize` is used for the base of the log, so it will result in the `problemScale` being less accurate (since instead of having values commonly be `0` or `1`, they be a decimal)

Reply:
There an issue with `valueCount`:

- Is it the number of distinct values defined by the problem? If so, to prevent double counting, we would need to put all the values into a set (value ranges can share values).
- Or is it the sum of possible assignments per variable?

Reply:
Yeah, I was worried about that. I don't think it's practical to put the items in a set. There could just be prohibitively too many. Double counting it is; the number is an _estimate_ and it needs to be made clear.